app/connectors/ITSAStatusConnector.scala

```diff
@@ -49,19 +51,48 @@ class ITSAStatusConnector @Inject()(val http: HttpClient,
         case OK =>
           response.json.validate[List[ITSAStatusResponseModel]].fold(
             invalid => {
-              Logger("application").error(s"[IncomeTaxViewChangeConnector][getITSAStatusDetail] - Json validation error parsing repayment response, error $invalid")
+              Logger("application").error(s"[ITSAStatusConnector][getITSAStatusDetail] - Json validation error parsing repayment response, error $invalid")
               Left(ITSAStatusResponseError(INTERNAL_SERVER_ERROR, "Json validation error parsing ITSA Status response"))
             },
             valid => Right(valid)
           )
         case status =>
           if (status >= INTERNAL_SERVER_ERROR) {
-            Logger("application").error(s"[IncomeTaxViewChangeConnector][getITSAStatusDetail] - Response status: ${response.status}, body: ${response.body}")
+            Logger("application").error(s"[ITSAStatusConnector][getITSAStatusDetail]xxx - Response status: ${response.status}, body: ${response.body}")
           } else {
-            Logger("application").warn(s"[IncomeTaxViewChangeConnector][getITSAStatusDetail] - Response status: ${response.status}, body: ${response.body}")
+            Logger("application").warn(s"[ITSAStatusConnector][getITSAStatusDetail] - Response status: ${response.status}, body: ${response.body}")
           }
           Left(ITSAStatusResponseError(response.status, response.body))
       }
     }
   }
+
+  def getOverwriteItsaStatusUrl(nino: String, taxYearRange: String, itsaStatus: String): String = {
+    s"${appConfig.itvcDynamicStubUrl}/income-tax-view-change/itsa-status/$nino/$taxYearRange/overwrite/$itsaStatus"
+  }
+
+
+  def overwriteItsaStatus(nino: Nino, taxYearRange: String, itsaStatus: String)
+                              (implicit headerCarrier: HeaderCarrier): Future[Unit] = {
+
+
+    http.GET[HttpResponse](getOverwriteItsaStatusUrl(nino.value, taxYearRange, itsaStatus))(
+      httpReads,
+      headerCarrier.withExtraHeaders("Accept" -> "application/vnd.hmrc.2.0+json"),
+      ec
+    ) flatMap { response =>
```

Comment:
there is no concern regarding usage of flatMap here;
Please provide specific reason to clarify why you think otherwise

Reply:
No specific reason, it looked redundant to create Futures in order to use a flatmap. Is there a benefit to using flatMap here?

Reply:
Future.fail and Future.successful has not implication on the existing thread pool;
first approach is more extendable in case we would like to use for-comprehension ie:

```
def overwriteCalculationList(nino: Nino, taxYearRange: String, crystallisationStatus: String)
                              (implicit headerCarrier: HeaderCarrier): Future[Unit] = {

    for {
      response <- http.GET[HttpResponse](getOverwriteCalculationListUrl(nino.value, taxYearRange, crystallisationStatus))(
        httpReads,
        headerCarrier.withExtraHeaders("Accept" -> "application/vnd.hmrc.2.0+json"),
        ec)
    } yield
      response.status match {
        case OK =>
          Future.successful((): Unit)
        case _ => Future.failed(new Exception(s"Overwrite unsuccessful. ~ Response status: ${response.status} ~. < Response body: ${response.body} >"))
      }
  } 
```